### PR TITLE
✨feat (spotify): Add spotify playlist and album search

### PR DESCRIPTION
**Pull request by dillionverma**
## Description

Add spotify playlist and album search

## Screencast

Screen Shot 2022-04-26 at 10 43 23 PM

Screen Shot 2022-04-26 at 10 43 45 PM

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by dillionverma** [comment]
This PR addresses #1507 #602 #910

**Comment by mattisssa** [comment]
Hey @dillionverma! Thanks for the contribution, I was expecting someone will tackle these requests. But actually, I recently made a quite bit update for the Spotify extension and switched it to the new OAuth + add commands you've made as well. The reason why it isn't in the Store yet is that it's under review by the Spotify team now and I guess it takes a while.

Have you seen my message in [the Slack Community](https://raycastcommunity.slack.com/archives/C02HEMAF2SJ/p1650620228132659) and [this PR](https://github.com/raycast/extensions/pull/1464)? 

I haven't done the **Search Playlist** command, I think what we can do here is to add your command on top of the PR. Or if you have different ideas, let me know 😉 

As I wrote in the message I can add you as a Tester in the extension so you could use the OAuth to play around.

**Comment by dillionverma** [comment]
Hey @mattisssa! That's a great idea. Using oauth would enable for controls specific to the user such as liking songs which would be cool.

I just hopped on the slack community and read over your PR and it looks good.

I would be happy to add the playlist command ontop of your PR. Another addition I did was modify the formatting for **tracks**. Here is an example of what it looks like:

Screen Shot 2022-04-27 at 11 15 08 AM

**Comment by mattisssa** [comment]
> Hey @mattisssa! That's a great idea. Using oauth would enable for controls specific to the user such as liking songs which would be cool.
> 
> I just hopped on the slack community and read over your PR and it looks good.
> 
> I would be happy to add the playlist command ontop of your PR. Another addition I did was modify the formatting for **tracks**. Here is an example of what it looks like:
> 
> Screen Shot 2022-04-27 at 11 15 08 AM

Hey! Sorry for the late reply, I actually was in contact with Spotify developers about my application in their system. They've rejected it for an unclear reason and still didn't respond to me. 

So yes, let's probably add your changes into the existing extension until I figure out how to pass the review 🤯

File: extensions/spotify-player/src/searchAlbums.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  const [searchText, setSearchText] = useState<string>();
+  const response = useAlbumSearch(searchText);
+
+  if (response.error) {
+    showToast(Toast.Style.Failure, "Search has failed", response.error);
+  }
+
+  return (
+    <List
+      searchBarPlaceholder="Search music by keywords..."
```
Comment (mattisssa) [inline extensions/spotify-player/src/searchAlbums.tsx:17]:
```suggestion
      searchBarPlaceholder="Search albums..."
```

File: extensions/spotify-player/src/searchPlaylists.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  const [searchText, setSearchText] = useState<string>();
+  const response = usePlaylistSearch(searchText);
+
+  if (response.error) {
+    showToast(Toast.Style.Failure, "Search has failed", response.error);
+  }
+
+  return (
+    <List
+      searchBarPlaceholder="Search music by keywords..."
```
Comment (mattisssa) [inline extensions/spotify-player/src/searchPlaylists.tsx:17]:
```suggestion
      searchBarPlaceholder="Search playlists..."
```

**Review by mattisssa — APPROVED** [review]

**Pull request merged**